### ISSUE TITLE
疑问 - 能仿人体属性识别模型PPLCNet_x1_0.yaml写一个 能实现多分类多标签新模型吗？ 

### ISSUE THREAD

**Issue by gg22mm** [PaddlePaddle/PaddleClas#2152]
疑问如下：

https://github.com/PaddlePaddle/PaddleClas/blob/release/2.4/docs/zh_CN/PULC/PULC_person_attribute.md

PULC 人体属性识别模型识别后：

[{'attributes': ['Male', 'Age18-60', 'Back', 'Glasses: False', 'Hat: False', 'HoldObjectsInFront: False', 'Backpack', 'Upper: LongSleeve UpperPlaid', 'Lower:  Trousers', 'No boots'], 'output': [0, 0, 0, 1, 0, 0, 1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 1, 0, 0, 1, 0, 0, 0, 0, 1]}]

其中： [0, 0, 0, 1, 0, 0, 1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 1, 0, 0, 1, 0, 0, 0, 0, 1]  各代表的是什么，跟前面的文字也对不上。

 如果想训练自己的多分类多标签能用吗？我自己能不能仿人体属性识别模型PPLCNet_x1_0.yaml写一个就能实现得到？  PaddleClas  的文档的二次开发新模型 能不能来个文档呢？

**Comment by tranngocphuong89** [comment]
好想法！
我也期待中~~~

**Comment by gg22mm** [comment]
> 好想法！ 我也期待中~~~

其实已经有大老在弄了，在只改配置的情况可以做多标签。  但是我上面提的：['Male', 'Age18-60', 'Back', 'Glasses: False', 'Hat: False', 'HoldObjectsInFront: False', 'Backpack', 'Upper: LongSleeve UpperPlaid', 'Lower: Trousers', 'No boots'], 'output': [0, 0, 0, 1, 0, 0, 1, 0, 0, 0, 0, 1, 0, 0, 0, 0, 0, 1, 0, 0, 1, 0, 0, 0, 0, 1]  他们不是  一 一 对应的，有点奇怪！

**Comment by gg22mm** [comment]
`
如果只是想做普通的多标签可以看这个：
基于PaddleClas的NUS-WIDE-SCENE多标签图像分类
			
			说明：https://aistudio.baidu.com/aistudio/projectdetail/4247343?channelType=0&channel=0
			
			数据查看
				其中第一列为图像文件名，其后36列分别为garden glacier grass harbor house lake moon mountain nighttime ocean plants railroad rainbow reflection road sky snow street sunset temple town valley water waterfall window 标签，为1则是，0否。
			
				最终预测结果： [{'class_ids': [6, 13, 23, 30], 'scores': [0.97452, 0.59816, 0.98675, 0.81546], 'file_name': './deploy/images/0517_2715693311.jpg', 'label_names': []}]
				
				即：clouds、lake、sky、water
		
			
			复制配置文件并修改
				配置文件为 PaddleClas/ppcls/configs/quick_start/professional/MobileNetV1_multilabel.yaml  改成  MobileNetV1_multilabel_wll_test.yaml
				
			
			
			.bug修复
				PaddleClas/ppcls/data/dataloader/multilabel_dataset.py的 label_ration 的bug修复，具体如下。
				
				# 这边判断依旧是，因为默认False，二用None判断会出错。
				# if self.label_ratio is not None:
				if self.label_ratio:
				
				
			训练：
				python  tools/train.py -c ./ppcls/configs/quick_start/professional/MobileNetV1_multilabel_wll_test.yaml
				
			
			模型评估
				python3 tools/eval.py -c ./ppcls/configs/quick_start/professional/MobileNetV1_multilabel_wll_test.yaml -o Arch.pretrained="./output/MobileNetV1/best_model"

				最终评估结果： [Eval][Epoch 0][Avg]MultiLabelLoss: 0.16364, loss: 0.16364, HammingDistance: 0.05834, AccuracyScore: 0.94166
			
			
			模型预测
				python3 tools/infer.py -c ./ppcls/configs/quick_start/professional/MobileNetV1_multilabel_wll_test.yaml -o Arch.pretrained="./output/MobileNetV1/best_model"	
				
				最终预测结果： [{'class_ids': [6, 13, 23, 30], 'scores': [0.97452, 0.59816, 0.98675, 0.81546], 'file_name': './deploy/images/0517_2715693311.jpg', 'label_names': []}]

				即：clouds、lake、sky、water
`

**Comment by tranngocphuong89** [comment]
> ` 如果只是想做普通的多标签可以看这个： 基于PaddleClas的NUS-WIDE-SCENE多标签图像分类
> 
> ```
> 		说明：https://aistudio.baidu.com/aistudio/projectdetail/4247343?channelType=0&channel=0
> 		
> 		数据查看
> 			其中第一列为图像文件名，其后36列分别为garden glacier grass harbor house lake moon mountain nighttime ocean plants railroad rainbow reflection road sky snow street sunset temple town valley water waterfall window 标签，为1则是，0否。
> 		
> 			最终预测结果： [{'class_ids': [6, 13, 23, 30], 'scores': [0.97452, 0.59816, 0.98675, 0.81546], 'file_name': './deploy/images/0517_2715693311.jpg', 'label_names': []}]
> 			
> 			即：clouds、lake、sky、water
> 	
> 		
> 		复制配置文件并修改
> 			配置文件为 PaddleClas/ppcls/configs/quick_start/professional/MobileNetV1_multilabel.yaml  改成  MobileNetV1_multilabel_wll_test.yaml
> 			
> 		
> 		
> 		.bug修复
> 			PaddleClas/ppcls/data/dataloader/multilabel_dataset.py的 label_ration 的bug修复，具体如下。
> 			
> 			# 这边判断依旧是，因为默认False，二用None判断会出错。
> 			# if self.label_ratio is not None:
> 			if self.label_ratio:
> 			
> 			
> 		训练：
> 			python  tools/train.py -c ./ppcls/configs/quick_start/professional/MobileNetV1_multilabel_wll_test.yaml
> 			
> 		
> 		模型评估
> 			python3 tools/eval.py -c ./ppcls/configs/quick_start/professional/MobileNetV1_multilabel_wll_test.yaml -o Arch.pretrained="./output/MobileNetV1/best_model"
> 
> 			最终评估结果： [Eval][Epoch 0][Avg]MultiLabelLoss: 0.16364, loss: 0.16364, HammingDistance: 0.05834, AccuracyScore: 0.94166
> 		
> 		
> 		模型预测
> 			python3 tools/infer.py -c ./ppcls/configs/quick_start/professional/MobileNetV1_multilabel_wll_test.yaml -o Arch.pretrained="./output/MobileNetV1/best_model"	
> 			
> 			最终预测结果： [{'class_ids': [6, 13, 23, 30], 'scores': [0.97452, 0.59816, 0.98675, 0.81546], 'file_name': './deploy/images/0517_2715693311.jpg', 'label_names': []}]
> 
> 			即：clouds、lake、sky、water
> ```
> 
> `

你分享的内容太好了。 感谢
